app/models/setting/discovered.rb

```diff
@@ -28,6 +28,8 @@ def self.load_defaults
         self.set('discovery_facts_hardware', N_("Regex to organize facts for hardware section"), "", N_("Hardware facts")),
         self.set('discovery_facts_network', N_("Regex to organize facts for network section"), "", N_("Network facts")),
         self.set('discovery_facts_ipmi', N_("Regex to organize facts for ipmi section"), "", N_("IPMI facts")),
+        self.set('discovery_lock', N_("Automatically generate PXE configuration to pin a newly discovered host to discovery"), false,N_("Lock PXE")),
+        self.set('discovery_lock_template', N_("PXE template to be used when pinning a host to discovery"), 'pxelinux_discovery',N_("PXE lock template name")),
```

Comment:
Oh sorry, folks have changed the API before they merged it. It's now a Proc, can you change it to (an example):

    ... { :collection => Proc.new {Hash[Environment.all.map{|env| [env[:name], env[:name]]}]} }

There is a missing space after comma above.

I also dont like the setting name, I suggest "Locked template name" or something similar?